pkg/controller/csi_snapshot_controller.go

```diff
@@ -0,0 +1,740 @@
+/*
```

Comment:
Why is this called `csi_snapshot_controller.go`. This looks like general binding logic, not very csi specific?

Reply:
You are right.  Initially we only had controller.go and csi_handler.go and later restructured the logic and modeled controller.go and csi_snapshot_controller.go after pv_controller_base.go and pv_controller.go.  Maybe we can name them snapshot_controller_base.go and snapshot_controller.go.

Reply:
This csi_snapshot_controller sits between the case controller and csi_handler, so we use the name csi. 
Logically, it follows pv_controller_base and pv_controller design pattern.